### DISCUSSION TITLE
Relay on Sonoff Basic automatically turns off after a second if rule for displaytext is set

### DISCUSSION THREAD

**Discussion by Rush56c** [arendst/Tasmota#16466] [Q&A] [answered]
Hi everyone.

I'm very new to all this, so far i was able to set everything up and running and configure as i wanted.

Now i got to a problem. I have 2 sonoff basic R2 switches. One is normally running with OLED display + temp/hum sensor with rules to turn on/off depending on humidity and displaytext with values on OLED display. 

Recently i got another display that i hooked to another switch and managed to set it up today. That switch has rules set to power on fan/switch every 10 minutes and turn off after 40 seconds. Now i added rule to display Temp/Hum values on display.

And it displays. BUT here is the problem. As soon as this rule is enabled, relay powers off after a second, whather triggered manually, by automation, anyhow. 

Could anyone help me? I can't find solution anywhere. 

Here are my rules:

Rule1 on Time#Minute|10 do backlog Power1 1 endon on Power1#state=1 do backlog Delay 400; power1 0 endon
Rule2 on SHT3X#Temperature do backlog Displaytext [f0s3] T [l1c3] %value% endon on SHT3X#Humidity do backlog Displaytext [l2c1] H [l2c3] %value% endon

Thanks..

**Comment by barbudor** [comment, marked as answer]
Don't use delay as there is no guarantee
Either use Pulsetime or RuleTimer

**Reply by Rush56c** [reply]
Thank you so much for a quick answer and a solution.  
Now i see why it worked on first switch, as i haven't used delay there.

Best Regards.
